samples/azure/azure-service-fabric-routing/sample.md

```diff
@@ -0,0 +1,207 @@
+---
+title: Azure Service Fabric Partition Aware Routing
+summary: Implementing partition aware routing for services hosted inside Service Fabric cluster.
+reviewed: 2017-02-14
+component: Core
+related:
+- nservicebus/azure-service-bus
+---
+
+The sample demonstrates how the NServiceBus extension points API can be used to implement partition aware routing for services hosted inside a Service Fabric cluster. It takes advantage of [routing system extensibility points](/nservicebus/messaging/routing-extensibility.md) and [custom pipeline behaviors](/nservicebus/pipeline/manipulate-with-behaviors.md) to support various types of NServiceBus communication patterns. It is assumed that the NServiceBus users are able to define mapping between message type and service partition for each command and event. It is also assumed that `send local`, `timeout` and `reply` messages are partition affine i.e. should be processed in the context of originating partition. The sample consists of services hosted inside Service Fabric as well as outside of the cluster and enables proper communication also in such scenario.
+
+## Prerequisites
+
+1. Strong understanding of Service Fabric [Reliable Services](https://docs.microsoft.com/en-us/azure/service-fabric/service-fabric-reliable-services-quick-start). 
+1. Service Fabric [development environment setup](https://docs.microsoft.com/en-us/azure/service-fabric/service-fabric-get-started) with dev cluster configured to run on 5 nodes.
+1. An Azure Service Bus namespace that can be used for communication between the instances.
+1. A system environment variable named "AzureServiceBus.ConnectionString" set to the connection string of the Azure Service Bus namespace. The connection string must provide Manage rights.
+
+NOTE: A Service Fabric cluster runs under the Network Service account and only reads system environment variables. Make sure the environment variable "AzureServiceBus.ConnectionString" is defined as a system environment variable and not user-scoped.
+
+NOTE: All services hosted inside Service Fabric custer log diagnostic information using [Event Tracing for Windows] (https://docs.microsoft.com/en-us/azure/service-fabric/service-fabric-diagnostics-how-to-monitor-and-diagnose-services-locally). In most of the cases no action is required to see application logs, however in some cases it might be necessary to add `MyCompany-ServiceFabricRouting-ZipCodeVoteCount` to the list of know [ETW providers](http://stackoverflow.com/a/35347603/2672802). 
```

Comment:
> @SeanFeldman Does that mean there is always just max one instance per partition?

One _active_ instance per partition. That's exactly it @ramonsmits.
There are also additional replicas in stand-by mode for HA, but those do not process messages.